pkg/dao/repository_configs.go

```diff
@@ -372,7 +372,34 @@ func (r repositoryConfigDaoImpl) filteredDbForList(OrgID string, filteredDB *gor
 
 	if filterData.Status != "" {
 		statuses := strings.Split(filterData.Status, ",")
-		filteredDB = filteredDB.Where("status IN ?", statuses)
+		var repoConfigs []models.RepositoryConfiguration
+
+		filteredDB = filteredDB.
+			Joins("LEFT JOIN tasks ON repository_configurations.last_snapshot_task_uuid = tasks.id").
+			Preload("Repository").
+			Preload("LastSnapshotTask").
+			Find(&repoConfigs)
+
+		var filteredRepoConfigsByStatus []models.RepositoryConfiguration
+
+		// filter repo configs by status(es)
+		for _, repoConfig := range repoConfigs {
+			combinedStatus := combineIntrospectionAndSnapshotStatuses(&repoConfig, &repoConfig.Repository)
+			for _, status := range statuses {
+				if combinedStatus == status {
+					filteredRepoConfigsByStatus = append(filteredRepoConfigsByStatus, repoConfig)
+				}
+			}
+		}
+
+		// retrieve all uuids from the filtered repo configs by status
+		var filteredRepoConfigUuids []string
+		for _, repoConfig := range filteredRepoConfigsByStatus {
+			filteredRepoConfigUuids = append(filteredRepoConfigUuids, repoConfig.UUID)
+		}
+
+		filteredDB = filteredDB.
+			Where("repository_configurations.uuid IN ?", filteredRepoConfigUuids)
```

Comment:
Is there any way to make this filter into a query?  I could see it quite complicated, and we'd have to essentially duplicate the combineIntrospectionAndSnapshotStatuses function to generate a where clause.  It think i'd be  something like this (this isn't complete though):

```
if filterData.Status == 'Pending' {
  return db.Where("LastIntrospectionStatus == 'pending' ").
                    Where( 
                             db.Where(LastSnapshotTask.STATUS is null").
                            Or("repoConfig.LastSnapshotTask.Status == 'running'") )
}
```

and it may be easier to just write the complicated Where() clause as pure sql if you can:

```
db.Where("LastIntrospectionStatus == 'pending' 
                     AND (LastSnapshotTask.STATUS is null OR 
                               repoConfig.LastSnapshotTask.Status == 'running')")
```

I think gorm allows that?

Reply:
and the reason to do this, is for performance.  It keeps the query in a single db call, rather than fetching having to fetch all results and re-query.  

Also there's a limit to the number of items you can put in an IN  clause

Reply:
ah yea that makes sense, i'll work on changing this to a query

Reply:
i've pushed up what i have so far for the query, still seeing incorrect results when filtering on multiple statuses though. with multiple statuses, the filter will only return the results for the first status in the list. i think it's the way i'm writing the where clauses maybe. let me know if you have ideas